File: programmingclub/traffic/traffic.java
import java.io.*;
import java.util.*;

public class traffic {
    public static void main(String[] args) throws IOException{
        Scanner in = new Scanner(new File("traffic.in"));
        PrintWriter out = new PrintWriter("traffic.out");

        int N = in.nextInt();
        String types[] = new String[N];
        int lbounds[] = new int[N]; // lower bounds
        int ubounds[] = new int[N]; // upper bounds

        for (int i = 0; i < N; i++) {
            types[i] = in.next();
            lbounds[i] = in.nextInt();
            ubounds[i] = in.nextInt();
        }

        int min = 0, max = 1000000;
        for (int i = N - 1; i >= 0; i--) { // Iterating in reverse to get the initial flow range
            if (types[i].equals("on")) {
                min = Math.max(min - ubounds[i], 0);
                max -= lbounds[i];
            } else if (types[i].equals("off")) {
                min += lbounds[i];
                max += ubounds[i];
            } else {
                min = Math.max(min, lbounds[i]);
                max = Math.min(max, ubounds[i]);
            }
        }
        out.println(min + " " + max);

        min = 0; max = 1000000;
        for (int i = 0; i < N; i++) { // Iterating normally to get the ending range
            if (types[i].equals("on")) {
                min += lbounds[i];
                max += ubounds[i];
            } else if (types[i].equals("off")) {
                min = Math.max(min - ubounds[i], 0);
                max -= lbounds[i];
            } else {
                min = Math.max(min, lbounds[i]);
                max = Math.min(max, ubounds[i]);
            }
        }
        out.println(min + " " + max);
        out.close();
    }
}

File: programmingclub/traffic/traffic.py
fin = open('traffic.in', 'r')
fout = open('traffic.out', 'w')

types = []
lbounds = [] # lower bounds
ubounds = [] # upper bounds
N = int(fin.readline())

for i in range(N):
    line = fin.readline().split()
    types.append(line[0])
    lbounds.append(int(line[1]))
    ubounds.append(int(line[2]))

minf = 0
maxf = 1000000;
for i in reversed(range(N)): # Iterating in reverse to get the initial flow range
    if types[i] == "on":
        minf = max(minf - ubounds[i], 0);
        maxf -= lbounds[i];
    elif types[i] == "off":
        minf += lbounds[i];
        maxf += ubounds[i];
    else:
        minf = max(minf, lbounds[i]);
        maxf = min(maxf, ubounds[i]);

fout.write(str(minf) + ' ' + str(maxf) + '\n')

minf = 0
maxf = 1000000;
for i in range(N): # Iterating normally to get the ending range
    if types[i] == "on":
        minf += lbounds[i];
        maxf += ubounds[i];
    elif types[i] == "off":
        minf = max(minf - ubounds[i], 0);
        maxf -= lbounds[i];
    else:
        minf = max(minf, lbounds[i]);
        maxf = min(maxf, ubounds[i]);

fout.write(str(minf) + ' ' + str(maxf))

fout.close()